src/boot/store.js

```diff
@@ -118,12 +119,57 @@ const migrations: { [string]: (GlobalState) => GlobalState } = {
       ackedPushToken: null,
     })),
   }),
+
+  '10': state => ({
+    ...state,
+    accounts: state.accounts.map(a => ({
+      ...a,
+      zulipVersion:
+        // $FlowMigrationFudge
+        a.zulipVersion instanceof ZulipVersion ? a.zulipVersion : new ZulipVersion(a.zulipVersion),
+    })),
+  }),
 };
 
+/**
+ * Persist ZulipVersion instances as raw version string.
+ */
+const zulipVersionTransform = createTransform(
+  // Only applies to the `zulipVersion` field in state.accounts.
+  (inboundState, key) => {
+    if (key !== 'accounts') {
+      return inboundState;
+    }
```

Comment:
Yes, it's redundant; see my comment [here](https://github.com/zulip/zulip-mobile/issues/3951#issuecomment-594790134):

> The createTransform API lets you whitelist some parts of the state; this example uses that, as well as redundant checks within the inbound and outbound transform functions, to show that it can be done either way or both ways.

Reply:
> If there's an accounts key at a lower level than the root (because we add another key with that name in some substate somewhere, or perhaps because someone's added a custom accounts emoji on their realm), will that key be processed?

> [...] skimming redux-persist indicates [...] "no"

Right.